muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/chat/chat.tsx

```diff
@@ -209,6 +237,123 @@ class Chat extends React.Component<IChatProps, IChatState> {
+
+      const studentsGuidanceCouncelors: SummaryStudentsGuidanceCouncelorsType[] =
+        [];
+      if (studentsUserGroups && studentsUserGroups.length) {
+        const councelGroups = studentsUserGroups.filter(
+          (studentsUserGroup) => studentsUserGroup.isGuidanceGroup == true
+        );
+
+        await Promise.all(
+          councelGroups.map(async (studentsUserGroup) => {
+            await promisify(
+              mApi().usergroup.groups.staffMembers.read(studentsUserGroup.id, {
+                properties:
+                  "profile-phone,profile-appointmentCalendar,profile-whatsapp,profile-vacation-start,profile-vacation-end",
+              }),
+              "callback"
+            )().then((result: SummaryStudentsGuidanceCouncelorsType[]) => {
+              result.forEach((studentsGuidanceCouncelor) => {
+                if (
+                  !studentsGuidanceCouncelors.some(
+                    (existingStudentCouncelor) =>
+                      existingStudentCouncelor.userEntityId ==
+                      studentsGuidanceCouncelor.userEntityId
+                  )
+                ) {
+                  studentsGuidanceCouncelors.push(studentsGuidanceCouncelor);
+                  studentsGuidanceCouncelors.sort((x, y) => {
+                    const a = x.lastName.toUpperCase(),
+                      b = y.lastName.toUpperCase();
+                    return a == b ? 0 : a > b ? 1 : -1;
+                  });
+                }
+              });
+            });
+          })
+        );
+      }
+      this.setState({
+        studyGuiders: studentsGuidanceCouncelors,
+      });
+    } catch (e) {
+      this.props.displayNotification(
+        this.props.i18n.text.get(
+          "plugin.chat.notification.councelorLoadFailed"
+        ),
+        "error"
+      );
+    }
+  }
+
+  /**
+   * getRoster gets roster from openfire and stores it in the component state
+   */
+  getRoster = async () => {
+    const stanza = $iq({
+      from: this.state.connection.jid,
+      type: "get",
+    }).c("query", { xmlns: Strophe.NS.ROSTER });
+
+    const jids: IChatContact[] = [];
+
+    const answerStanza: Element = await new Promise((resolve) => {
+      this.state.connection.sendIQ(stanza, (answerStanza: Element) => {
+        resolve(answerStanza);
+      });
+    });
+
+    const rosterStanza = answerStanza.querySelectorAll("query item");
+
+    rosterStanza.forEach((r) => {
+      const jId = r.getAttribute("jid");
+      jids.push({ jid: jId });
+    });
+
+    if (jids.length > 0) {
+      const chatRoster: IChatContact[] = [];
+      await Promise.all(
+        jids.map(async (contact: IChatContact) => {
+          await promisify(
+            mApi().chat.userInfo.read(contact.jid.split("@")[0], {}),
+            "callback"
+          )().then((user: IChatContact) => {
+            chatRoster.push({
+              ...{ jid: contact.jid },
+              ...user,
+            });
+          });
```

Comment:
Here you are adding new item to array. To clear this a bit, spread object first and then change jid to what you want it to really be.
```ts
chatRoster.push({...user, jid: contact.jid});

```